src/libstd/path.rs

```diff
@@ -639,6 +639,18 @@ pub struct Iter<'a> {
     inner: Components<'a>,
 }
 
+#[stable(feature = "path_components_debug", since = "")]
+impl<'a> fmt::Debug for Components<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let components = self.as_path()
+                             .components()
+                             .collect::<Vec<_>>();
```

Comment:
Yeah, I can switch this to `debug_list` to avoid the allocation, though we lose the "Components" title. I wonder if there is value in adding a `debug_iter` that offers `name` and `iterator` params.